pkg/store/employee/util.go

```diff
@@ -11,27 +11,30 @@ func getByWhereConditions(query *gorm.DB, filter EmployeeFilter) *gorm.DB {
 	}
 
 	if len(filter.Positions) > 0 {
-		query = query.Joins("JOIN employee_positions ON employees.id = employee_positions.employee_id JOIN positions ON employee_positions.position_id = positions.id AND positions.code IN ?",
-			filter.Positions)
+		if filter.Positions[0] == "-" {
+			query = query.Joins(`LEFT JOIN employee_positions ON employees.id = employee_positions.employee_id AND employee_positions.id IS NULL`)
+		} else {
+			query = query.Joins("JOIN employee_positions ON employees.id = employee_positions.employee_id JOIN positions ON employee_positions.position_id = positions.id AND positions.code IN ?",
+				filter.Positions)
+		}
+
 	}
 
 	if len(filter.Stacks) > 0 {
-		query = query.Joins("JOIN employee_stacks ON employees.id = employee_stacks.employee_id JOIN stacks ON employee_stacks.stack_id = stacks.id AND stacks.code IN ?",
-			filter.Stacks)
+		if filter.Stacks[0] == "-" {
+			query = query.Joins(`LEFT JOIN employee_stacks ON employees.id = employee_stacks.employee_id AND employee_stacks.id IS NULL`)
+		} else {
+			query = query.Joins("JOIN employee_stacks ON employees.id = employee_stacks.employee_id JOIN stacks ON employee_stacks.stack_id = stacks.id AND stacks.code IN ?",
+				filter.Stacks)
+		}
 	}
 
 	if len(filter.Projects) > 0 {
 		if filter.Projects[0] == "-" {
-			query = query.Where(`employees.id NOT IN (
-				SELECT project_members.employee_id 
-				FROM project_members
-				WHERE project_members.deleted_at IS NULL AND project_members.status = 'active')`)
+			query = query.Joins(`LEFT JOIN project_members ON employees.id = project_members.employee_id AND project_members.id IS NULL`)
```

Comment:
duplicate employees

Reply:
cannot be duplicated because can just have only 1 record `employee` with no `project_members`